restapi/src/test/java/io/stargate/web/docsapi/service/QueryExecutorTest.java

```diff
@@ -205,8 +205,10 @@ void testPartialFullScan(int pageSize) {
   }
 
   @ParameterizedTest
-  @CsvSource({"1", "3", "5", "100"})
+  @CsvSource({"3", "5", "100"})
   void testFullScanFinalPagingState(int pageSize) {
+    // Note: page size must not be a divisor of the result set size for this test to work normally
+    // Otherwise, the last doc will have a non-null paging state
```

Comment:
Does this have implications for actual usage or is this only a testing consideration?

Reply:
This is just to be able to do asserts on the paging state in a simple way. If the page size aligns exactly with the result set size, there will be an extra empty page fetched. We can account for that in the test, but it'll complicate the assertion logic.

In real applications this is not likely, I guess. Plus our default page size is 1000 rows, so fetching one more empty page at the end should not cause too much overhead.

However, now that you mention it, I think it might be worth splitting the test methods so that we validate paging state separately from doc groupling since pages size should not matter for the latter.... I'll see if that's too much refactoring.

Reply:
Updated tests. @dougwettlaufer : is the new test code clearer? I hope it is :)